src/main/scala/org/broadinstitute/hail/driver/GroupTestSKATO.scala

```diff
@@ -0,0 +1,359 @@
+    var missingCutoff: Double = 0.15
+
+    @Args4jOption(required = false, name = "--estimate-maf",
+      usage = "Method for estimating MAF. 1 = use all samples to estimate MAF. 2 = use samples with non-missing phenotypes and covariates")
+    var estimateMAF: Int = 1
+
+  }
+
+  def newOptions = new Options
+
+  def name = "grouptest skato"
+
+  def description = "Run SKAT-O on groups specified by a variant annotation"
+
+  def supportsMultiallelic = false
+
+  def requiresVDS = true
+
+  val skatoElementSignature = TStruct(
+    "groupName" -> TInt,
+    "pValue" -> TDouble,
+    "pValueNoAdj" -> TDouble,
+    "nMarker" -> TInt,
+    "nMarkerTest" -> TInt
+  )
+
+  val skatoSignature = TArray(skatoElementSignature)
+
+  val header = "groupName\tpValue\tpValueNoAdj\tnMarker\tnMarkerTest"
+
+  def run(state: State, options: Options): State = {
+    val vds = state.vds
+    val sampleIds = vds.sampleIds
+
+    val properties = try {
+      val p = new Properties()
+      val is = new FileInputStream(options.config)
+      p.load(is)
+      is.close()
+      p
+    } catch {
+      case e: IOException =>
+        fatal(s"could not open file: ${e.getMessage}")
+    }
+
+    val yType = if (options.quantitative) "C" else "D"
+    val nResampling = options.nResampling
+    val typeResampling = options.typeResampling
+    val adjustment = !options.noAdjustment
+    val kernel = options.kernel
+    val method = options.method
+    val weightsBeta = options.weightsBeta
+    val imputeMethod = options.imputeMethod
+    val rCorr = options.rCorr
+    val missingCutoff = options.missingCutoff
+    val estimateMAF = options.estimateMAF
+    val quantitative = options.quantitative
+    val splat = options.splat
+    val groupNameSep = options.groupNameSep
+
+    val R = properties.getProperty("hail.skato.Rscript", "/usr/bin/Rscript")
+    val skatoScript = properties.getProperty("hail.skato.script", "src/dist/scripts/skato.r")
+
+    val rCorrArray = options.rCorr.split(",").map(_.toDouble)
+    if (!rCorrArray.forall(d => d >= 0.0 && d <= 1.0))
+      fatal(s"Arguments to --r-corr must be numeric, between 0.0 and 1.0, and comma-separated. Found ${options.rCorr}")
+
+    val weightBetaArray = options.weightsBeta.split(",").map(_.toDouble)
+    if (weightBetaArray.length != 2)
+      fatal(s"Must give two comma-separated arguments to --weights-beta. Found ${options.weightsBeta}")
+
+    val availMethods = Set("davies", "liu", "liu.mod", "optimal.adj", "optimal")
+    if (!availMethods.contains(method))
+      fatal("Did not recognize option specified for --method. Use one of [davies, liu, liu.mod, optimal.adj, optimal]")
+
+    val availKernels = Set("linear", "linear.weighted", "IBS", "IBS.weighted", "quadratic", "2wayIX")
+    if (!availKernels.contains(kernel))
+      fatal("Did not recognize option specified for --kernel. Use one of [linear, linear.weighted, IBS, IBS.weighted, quadratic, 2wayIX]")
```

Comment:
use s"...[${availKernels.mkString(", ")}]..." instead of writing out the list again.